### PR TITLE
📖 add documentation about namingStrategy in ClusterClasses

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Follow-up to #9340 

Adds a section in the book about naming strategy in clusterclasses.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area book

### CODE REVIEW

File: docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md [hunk 1]
```diff
@@ -289,6 +292,103 @@ a default value, the value is automatically added to the variables list.
 
 </aside>
 
+## ClusterClass with custom naming strategies
+
+The controller needs to generate names for new objects when a Cluster is getting created
+from a ClusterClass. These names have to be unique on a per namespace base which mostly is
+accomplished by concatenating the cluster name in some way with a random suffix.
```
Comment (killianmuldoon) [inline docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md:299]:
```suggestion
from a ClusterClass. These names have to be unique for each namespace. The naming strategy enables this  by concatenating the cluster name with a random suffix.
```

**Review by killianmuldoon — COMMENTED** [review]
Looks good - one nit / clarification.

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

Thanks!

**Review by fabriziopandini — COMMENTED** [review]
Thanks for writing this doc!
/lgtm

**Pull request merged**